### ISSUE TITLE
Unhelpful error message when git branch does not exist.

### ISSUE THREAD

**Issue by neilbmclaughlin** [Azure/azure-cli#1651]
I had problems with failed function app deployments which had a sourcecontrols resource.

The branch name was of the form 'feature/do-something-new' which I eventually realised needed to be escaped (i.e. "branch": "[replace(variables('branch'), '/', '\/')]",). I was also able to replicate the deployment error by using a branch name which didn't exist (masterxxx).

My issue is specifically with the error message appearing in the activity log in the portal which was the uninformative:

```
        "statusCode": "Conflict",
        "serviceRequestId": "716e6819-cdfe-4fc5-b34e-d1528f1f4b60",
        "statusMessage": "{\"status\":\"Failed\",\"error\":{\"code\":\"ResourceDeploymentFailure\",\"message\":\"The resource operation completed with terminal provisioning state 'Failed'.\"}}"
```

Interestingly when opening the function app in the portal after the failed deployment with the unescaped 'feature/do-something-new' branch name the source code and functions were visible but not when opening the app following deployment using the unknown branch masterxxx.

The working template can be seen [here](https://github.com/nhsuk/service-etl-function-app/pull/14)

**Comment by neilbmclaughlin** [comment]
I'm not convinced it is a escaping character issue now as I have failing deployments even with the code change described above.

When looking at the continuous integration details for the function app I discovered some further logging for the failed github deployment:

```
Command: "D:\home\site\deployments\tools\deploy.cmd"
Handling node.js deployment.
KuduSync.NET from: 'D:\home\site\repository' to: 'D:\home\site\wwwroot'
Copying file: '.eslintignore'
Copying file: '.eslintrc.json'
Copying file: '.gitignore'
Copying file: '.istanbul.yml'
Copying file: '.tern-project'
Copying file: 'host.json'
Copying file: 'package.json'
Copying file: 'README.md'
Copying file: 'A_SplitFile\function.json'
Copying file: 'A_SplitFile\index.js'
Copying file: 'A__A_ProcessFirstSplitFile\function.json'
Copying file: 'A__A_ProcessFirstSplitFile\index.js'
Copying file: 'A__A_ProcessSecondSplitFile\function.json'
Copying file: 'A__A_ProcessSecondSplitFile\index.js'
Copying file: 'B_GenerateUrl\function.json'
Copying file: 'B_GenerateUrl\index.js'
Copying file: 'C_Download\function.json'
Copying file: 'C_Download\index.js'
Copying file: 'D_TransformAndStore\function.json'
Copying file: 'D_TransformAndStore\index.js'
Copying file: 'D_TransformAndStore\transform.js'
Copying file: 'scripts\build.sh'
Copying file: 'scripts\edispensary-50.csv'
Copying file: 'scripts\template.json'
Copying file: 'test\unit\transform.js'
Looking for app.js/server.js under site root.
Node.js versions available on the platform are: 0.6.20, 0.8.2, 0.8.19, 0.8.26, 0.8.27, 0.8.28, 0.10.5, 0.10.18, 0.10.21, 0.10.24, 0.10.26, 0.10.28, 0.10.29, 0.10.31, 0.10.32, 0.10.40, 0.12.0, 0.12.2, 0.12.3, 0.12.6, 4.0.0, 4.1.0, 4.1.2, 4.2.1, 4.2.2, 4.2.3, 4.2.4, 4.3.0, 4.3.2, 4.4.0, 4.4.1, 4.4.6, 4.4.7, 4.5.0, 4.6.0, 4.6.1, 5.0.0, 5.1.1, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.7.0, 5.7.1, 5.8.0, 5.9.1, 6.0.0, 6.1.0, 6.2.2, 6.3.0, 6.5.0, 6.6.0, 6.7.0, 6.9.0, 6.9.1, 6.9.2, 7.0.0, 7.1.0, 7.2.0, 7.3.0.
Selected node.js version 7.3.0. Use package.json file to choose a different version.
Selected npm version 3.10.10
Updating iisnode.yml at D:\home\site\wwwroot\iisnode.yml
Missing server.js/app.js files, web.config is not generated
Thread was being aborted.
Thread was being aborted.
```

Interestingly if I then hit the Redeploy button in the portal it is successful

```
Using cached version of deployment script (command: 'azure -y --no-dot-deployment -r "D:\home\site\repository" -o "D:\home\site\deployments\tools" --node --sitePath "D:\home\site\repository"').

Command: "D:\home\site\deployments\tools\deploy.cmd"
Handling node.js deployment.
KuduSync.NET from: 'D:\home\site\repository' to: 'D:\home\site\wwwroot'
Looking for app.js/server.js under site root.
Missing server.js/app.js files, web.config is not generated
Node.js versions available on the platform are: 0.6.20, 0.8.2, 0.8.19, 0.8.26, 0.8.27, 0.8.28, 0.10.5, 0.10.18, 0.10.21, 0.10.24, 0.10.26, 0.10.28, 0.10.29, 0.10.31, 0.10.32, 0.10.40, 0.12.0, 0.12.2, 0.12.3, 0.12.6, 4.0.0, 4.1.0, 4.1.2, 4.2.1, 4.2.2, 4.2.3, 4.2.4, 4.3.0, 4.3.2, 4.4.0, 4.4.1, 4.4.6, 4.4.7, 4.5.0, 4.6.0, 4.6.1, 5.0.0, 5.1.1, 5.3.0, 5.4.0, 5.5.0, 5.6.0, 5.7.0, 5.7.1, 5.8.0, 5.9.1, 6.0.0, 6.1.0, 6.2.2, 6.3.0, 6.5.0, 6.6.0, 6.7.0, 6.9.0, 6.9.1, 6.9.2, 7.0.0, 7.1.0, 7.2.0, 7.3.0.
Selected node.js version 7.3.0. Use package.json file to choose a different version.
Selected npm version 3.10.10
Updating iisnode.yml at D:\home\site\wwwroot\iisnode.yml
Finished successfully.
```

**Comment by neilbmclaughlin** [comment]
I'm pretty sure that this error was happening because of a missing dependsOn for sourcecontrols - it seems to have needed both Sites and Sites/config declared as dependencies. Closing this in view of my findings.

Error message could have been more helpful though :)
